usaspending_api/awards/management/commands/load_subawards.py

```diff
@@ -0,0 +1,193 @@
+import logging
+import os
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from functools import partial
+from psycopg2.sql import Identifier, Literal, SQL
+from usaspending_api.common.helpers.sql_helpers import convert_composable_query_to_string, get_connection
+from usaspending_api.common.helpers.timing_helpers import Timer
+
+
+logger = logging.getLogger("console")
+
+
+# All of our timers log to logger.info and logger.error.
+Timer = partial(Timer, success_logger=logger.info, failure_logger=logger.error)
+
+
+class Command(BaseCommand):
+    help = "Load subawards from Broker into USAspending."
+
+    def __init__(self, *args, **kwargs):
+        self.full_reload = False
+        self.log_sql = False
+        super().__init__(*args, **kwargs)
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--full-reload",
+            action="store_true",
+            help="Empties the USAspending subaward and broker_subaward tables before loading."
+        )
+
+        parser.add_argument(
+            "--sql",
+            action="store_true",
+            help="For debugging.  Log all SQL statements.  More verbose than you might expect."
+        )
+
+    def handle(self, *args, **options):
+
+        # Pick out just the last bit of the module name (which should match the
+        # file name minus the extension).
+        module_name = __name__.split(".")[-1]
+        with Timer(module_name):
+
+            self.full_reload = options["full_reload"]
+            self.log_sql = options["sql"]
+
+            if self.full_reload:
+                logger.info("--full-reload flag supplied.  Performing a full reload.")
+                where = ''
+
+            else:
+                max_id, max_created_at, max_updated_at = self._get_maxes()
+
+                if max_id is None and max_created_at is None and max_updated_at is None:
+                    logger.info("USAspending's broker_subaward table is empty.  Promoting to full reload.")
+                    self.full_reload = True
+                    where = ''
+
+                else:
+                    logger.info(
+                        "Performing an incremental load on USAspending's broker_subaward "
+                        "where id > {}, created_at > {}, or "
+                        "updated_at > {}".format(max_id, max_created_at, max_updated_at)
+                    )
+                    where = self._build_where(id=max_id, created_at=max_created_at, updated_at=max_updated_at)
+
+            self._perform_load(where)
+
+    def _execute_sql(self, sql):
```

Comment:
Stupid question: aren't similar `_execute_sql` and `_execute_sql_file` helper functions available? Totally optional given that these look like they include specific logs.

Reply:
Yes.  Similar.  But, as per usual, they didn't do quite what I wanted.